### PR TITLE
Create E0748 error code for unterminated raw strings

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Reopening of #66035.

r? @estebank

### CODE REVIEW

**Review by Xanewok — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
cc @rust-lang/compiler

File: src/librustc_error_codes/error_codes/E0747.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+Erroneous code example:
+
+```compile_fail,E0747
+let dolphins = r##"Dolphins!"#; // error!
+```
+To terminate a raw string, you have to have the same number of `#` at the end
+than at the beginning. Example:
+```
+let dolphins = r#"Dolphins!"#; // one `#` at the beginning, one at the end so
+                               // all good!
```
Comment (varkor) [inline src/librustc_error_codes/error_codes/E0747.md]:
```suggestion
let dolphins = r#"Dolphins!"#; // One `#` at the beginning; one at the end:
                               // all good!
```

**Review by varkor — APPROVED** [review]
r=me with tweaked sentence.

**Review by estebank — APPROVED** [review]

**Comment by estebank** [comment]
also r=me after applying varkor's changes

**Comment by GuillaumeGomez** [comment]
E0747 has been taken so let's move to E0748...

**Pull request merged**